prompting/tools/datasets/wiki.py

```diff
@@ -77,13 +72,32 @@ def _get_random_titles(pages=10, seed=42) -> List:
 
 
 @lru_cache(maxsize=1000)
-def _wiki_search(name, results) -> List:
+def _wiki_search(name: str, results: wiki.WikipediaPage) -> List:
     """Cached Wikipedia search."""
     return wiki.search(name, results=results)
 
+def get_article_sections(title: str) -> Dict:
+    # Fetch the HTML content of the Wikipedia article
+    url = f"https://en.wikipedia.org/wiki/{title}"
+    response = requests.get(url)
+    html_content = response.text
+
+    # Parse the HTML using BeautifulSoup
+    soup = BeautifulSoup(html_content, 'html.parser')
+
+    # Find all the sections and their content
+    sections = {}
+    for section in soup.find_all('h2'):
+        section_title = section.text
+        section_content_tag = section.find_next('p')
+        if section_content_tag is not None:
+            section_content = section_content_tag.text
+            sections[section_title] = section_content
+
+    return sections
 
 def process_page(
-    page, valid_header: callable = None, valid_content: callable = None
+    page, exclude_sections: list = None, valid_section: callable = None, selector: Selector = None
```

Comment:
Best to type as Optional[list] or list | None if you're allowing values to be nullable